symfony/plugins/orangehrmMaintenancePlugin/Service/PurgeEmployeeService.php

```diff
@@ -0,0 +1,139 @@
+     * @throws TransactionException
+     */
+    public function purgeEmployeeData(int $empNumber): void
+    {
+        $this->beginTransaction();
+        try {
+            $purgeableEntities = $this->getPurgeableEntities(self::GDPR_PURGE_EMPLOYEE);
+            foreach ($purgeableEntities as $purgeableEntityClassName => $purgeStrategies) {
+                foreach ($purgeStrategies['PurgeStrategy'] as $strategy => $strategyInfoArray) {
+                    $purgeStrategy = $this->getPurgeStrategy(
+                        $purgeableEntityClassName,
+                        $strategy,
+                        $strategyInfoArray
+                    );
+                    $purgeStrategy->purge($empNumber);
+                }
+            }
+            $this->commitTransaction();
+        } catch (Exception $exception) {
+            $this->rollBackTransaction();
+            throw new TransactionException($exception);
+        }
+
+    }
+
+
+//        $connection = Doctrine_Manager::getInstance()->getCurrentConnection();
+//        try {
+//            $connection->beginTransaction();
+//            $purgeableEntities = $this->getPurgeableEntities(self::EMPLOYEE_GDPR);
+//            foreach ($purgeableEntities as $purgeableEntityClassName => $purgeStrategies) {
+//                foreach ($purgeStrategies['PurgeStrategy'] as $strategy => $strategyInfoArray) {
+//                    $strategy = $this->getPurgeStrategy($purgeableEntityClassName, $strategy, $strategyInfoArray);
+//                    $strategy->purge($empNumber);
+//                }
+//            }
+//            $connection->commit();
+//        } catch (Exception $e) {
+//            $connection->rollback();
+//            Logger::getLogger('maintenance')->error($e->getCode() . ' - ' . $e->getMessage(), $e);
+//            throw new Exception($e->getMessage(), $e->getCode(), $e);
+//        }
+
+
+    /**
+     * @param string $fileName
+     * @return array
+     */
+    public function getPurgeableEntities(string $fileName): array
+    {
+        if (!isset($this->purgeableEntities)) {
+            $path = realpath(dirname(__FILE__, 2)) . '/config/' . $fileName . '.yaml';
+            $this->purgeableEntities = Yaml::parseFile($path);
+        }
+        return $this->purgeableEntities['Entities'];
+    }
+
+    /**
+     * @param string $purgeableEntityClassName
+     * @param string $strategy
+     * @param array $strategyInfoArray
+     * @return PurgeStrategy
+     */
+    public function getPurgeStrategy(string $purgeableEntityClassName, string $strategy, array $strategyInfoArray): PurgeStrategy
+    {
+        $purgeStrategyClass = 'OrangeHRM\\Maintenance\\PurgeStrategy\\' . $strategy . "PurgeStrategy";
+        return new $purgeStrategyClass($purgeableEntityClassName, $strategyInfoArray);
+    }
+
+    /**
+     * @param array $matchByValues
+     * @param string $table
+     * @return array
+     * LIKE A WRAPPER FUNCTION REMOVE LATER MAYBE
```

Comment:
will remove if no usage

Reply:
can get same function directly from Dao. Shall I remove this?